suite-native/tokens/src/tokensSelectors.ts

```diff
@@ -1,70 +1,249 @@
+    selectAccountByKey,
+    selectAccountTransactions,
+    selectVisibleDeviceAccountsByNetworkSymbol,
+    TransactionsRootState,
+} from '@suite-common/wallet-core';
+import {
+    AccountKey,
+    TokenAddress,
+    TokenInfoBranded,
+    TokenSymbol,
+} from '@suite-common/wallet-types';
+import { TokenInfo, TokenTransfer } from '@trezor/blockchain-link';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+
 import { isCoinWithTokens } from './utils';
+import { TypedTokenTransfer, WalletAccountTransaction } from './types';
 
 export type TokensRootState = AccountsRootState & DeviceRootState & TokenDefinitionsRootState;
 
-export const selectHasDeviceAnyTokens = (state: TokensRootState, coin: NetworkSymbol) => {
-    if (!isCoinWithTokens(coin)) return false;
+export const selectAccountTokenInfo = memoizeWithArgs(
+    (
+        state: AccountsRootState,
+        accountKey?: AccountKey,
+        tokenAddress?: TokenAddress,
+    ): TokenInfoBranded | null => {
+        const account = selectAccountByKey(state, accountKey);
+        if (!account || !account.tokens) {
+            return null;
+        }
+
+        return (
+            (A.find(
+                account.tokens,
+                (token: TokenInfo) => token.contract === tokenAddress,
+            ) as TokenInfoBranded) ?? null
+        );
+    },
+    // 100 is a reasonable size for the cache
+    { size: 100 },
+);
+
+export const selectAccountTokenSymbol = (
+    state: AccountsRootState,
+    accountKey?: AccountKey,
+    tokenAddress?: TokenAddress,
+): TokenSymbol | null => {
+    const tokenInfo = selectAccountTokenInfo(state, accountKey, tokenAddress);
+
+    if (!tokenInfo) {
+        return null;
+    }
+
+    // FIXME: This is the only place in the codebase where we change case of token symbol.
+    // The `toUpperCase()` operation is necessary because we are receiving wrongly formatted token symbol from connect.
+    // Can be removed at the moment when desktop issue https://github.com/trezor/trezor-suite/issues/8037 is resolved.
+    return tokenInfo.symbol.toUpperCase() as TokenSymbol;
+};
+
+export const selectAccountTokenTransactions = memoizeWithArgs(
+    (
+        state: TransactionsRootState,
+        accountKey: AccountKey,
+        tokenAddress: TokenAddress,
+    ): WalletAccountTransaction[] =>
+        pipe(
+            selectAccountTransactions(state, accountKey),
+            A.map(transaction => ({
+                ...transaction,
+                tokens: transaction.tokens.map((tokenTransfer: TokenTransfer) => ({
+                    ...tokenTransfer,
+                    symbol: tokenTransfer.symbol,
+                })),
+            })),
+            A.filter(transaction =>
+                A.some(
+                    transaction.tokens,
+                    tokenTransfer => tokenTransfer.contract === tokenAddress,
+                ),
+            ),
+        ) as WalletAccountTransaction[],
+    { size: 500 },
```

Comment:
I don't think we need such a big number, something like 50 will be also fine there because there won't more than one mounted instance of component that uses this selector

Reply:
👍  Will change. I kept the number from renamed and moved ethereum selector.

Reply:
e8dde2170e